pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2,301 +2,1132 @@ package listusers
 
 import (
 	"context"
+	"fmt"
 	"testing"
 
-	parser "github.com/craigpastro/openfga-dsl-parser/v2"
-	"github.com/google/go-cmp/cmp"
 	"github.com/oklog/ulid/v2"
 	openfgav1 "github.com/openfga/api/proto/openfga/v1"
 	"github.com/openfga/openfga/pkg/storage/memory"
+	"github.com/openfga/openfga/pkg/testutils"
 	"github.com/openfga/openfga/pkg/tuple"
 	"github.com/openfga/openfga/pkg/typesystem"
 	"github.com/stretchr/testify/require"
-	"google.golang.org/protobuf/protoadapt"
-	"google.golang.org/protobuf/testing/protocmp"
+	"go.uber.org/goleak"
+	"google.golang.org/protobuf/types/known/structpb"
 )
 
-func TestListUsers(t *testing.T) {
+type ListUsersTests []struct {
+	name                  string
+	TemporarilySkipReason string // Temporarily skip test until functionality is fixed
+	req                   *openfgav1.ListUsersRequest
+	model                 string
+	tuples                []*openfgav1.TupleKey
+	expectedUsers         []string
+	expectedError         error
+}
 
-	storeID := ulid.Make().String()
-	modelID := ulid.Make().String()
+func TestListUsersDirectRelationship(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	model := `model
+	schema 1.1
+	type user
+	type document
+		relations
+			define viewer: [user]`
 
-	tests := []struct {
-		name            string
-		req             *openfgav1.ListUsersRequest
-		typeDefinitions []*openfgav1.TypeDefinition
-		tuples          []*openfgav1.TupleKey
-		expectedUsers   []*openfgav1.User
-		expectedError   error
-	}{
+	tests := ListUsersTests{
```

Comment:
Please add 2 more test cases:

1. One where the filter's relation is inapplicable. e.g.

```
type user

type document
  relations
    define viewer: [user]
```
`ListUsers(document:1#viewer, user_filters=[{"type": "document", "relation": "undefined"}]`

2. Direct relationships with userset subjects and a userset filter. e.g.

```
type user

type group
  relations
    define member: [user, group#member]
```

- group:eng#member@group:fga#member
- group:fga#member@group:fga-backend#member

`ListUsers(group:eng#member, user_filters=[{"type": "group", "relation": "member"}]` should return ["group:eng#member", "group:fga#member", "group:fga-backend#member"]